k-distribution/include/kframework/builtin/domains.md

```diff
@@ -612,6 +612,93 @@ that two different sets will return the same element, even if they are similar.
+  // rule #Ceil(@S1:Set @S2:Set) =>
+  //        {intersectSet(@S1, @S2) #Equals .Set} #And #Ceil(@S1) #And #Ceil(@S2)
+  //   [simplification]
+
+//simpler version
+  rule #Ceil(@S:Set SetItem(@E:KItem)) =>
+         {(@E in @S) #Equals false} #And #Ceil(@S) #And #Ceil(@E)
+    [simplification]
+
+  // -Set simplifications
+  rule S              -Set .Set           => S          [simplification]
+  rule .Set           -Set  _             => .Set       [simplification]
+  rule SetItem(X)     -Set (S SetItem(X)) => .Set
+                               ensures notBool (X in S) [simplification]
+  rule S              -Set (S SetItem(X)) => .Set
+                               ensures notBool (X in S) [simplification]
+  rule (S SetItem(X)) -Set S              => SetItem(X)
+                               ensures notBool (X in S) [simplification]
+  rule (S SetItem(X)) -Set SetItem(X)     => S
+                               ensures notBool (X in S) [simplification]
+  // rule SetItem(X)     -Set S              => SetItem(X)
+  //                            requires notBool (X in S)  [simplification]
+  // rule (S1 SetItem(X)) -Set (S2 SetItem(X))  => S1 -Set S2
+  //                             ensures notBool (X in S1)
+  //                             andBool notBool (X in S2) [simplification]
+
+
+
+  // |Set simplifications
+  rule S    |Set .Set => S    [simplification]
+  rule .Set |Set S    => S    [simplification]
+  rule S    |Set S    => S    [simplification]
+
+  rule (S SetItem(X)) |Set SetItem(X) => S SetItem(X)
+                             ensures notBool (X in S) [simplification]
+//  rule (S SetItem(X)) |Set S          => S SetItem(X)
+//                             ensures notBool (X in S) [simplification]
+  rule SetItem(X) |Set (S SetItem(X)) => S SetItem(X)
+                             ensures notBool (X in S) [simplification]
+  rule S          |Set (S SetItem(X)) => S SetItem(X)
+                             ensures notBool (X in S) [simplification]
+
+  // intersectSet simplifications
+  rule intersectSet(.Set, _   ) => .Set    [simplification]
+  rule intersectSet( _  , .Set) => .Set    [simplification]
+  rule intersectSet( S  , S   ) =>  S      [simplification]
+
+  rule intersectSet( S SetItem(X), SetItem(X))     => SetItem(X)
+                                                        ensures notBool (X in S)      [simplification]
+  // rule intersectSet( S SetItem(X) , S)             => S ensures notBool (X in S)      [simplification]
+  rule intersectSet( S1 SetItem(X), S2 SetItem(X)) => intersectSet(S1, S2) SetItem(X)
+                                                        ensures notBool (X in S1)
+                                                        andBool notBool (X in S2)     [simplification]
+
+  // membership simplifications
+  rule E in .Set           => false requires #Ceil(E)       [simplification]
+  rule E in SetItem(E)     => true  requires #Ceil(E)       [simplification]
```

Comment:
You do not need the side condition here because E is an element variable and will not match \bottom regardless.

Reply:
> If you're removing this you need to add an implementation of this hook to the Haskell backend.

Good point, thank you. Looking back at #2689 I find that I can instead add `[concrete]` to avoid the rule in symbolic simplifications.

Removing the redundant `#Ceil`.